openeo/extra/job_management/stac_job_db.py

```diff
@@ -0,0 +1,300 @@
+import concurrent
+import datetime
+import logging
+from typing import Iterable, List, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import pystac
+import requests
+from pystac import Collection, Item
```

Comment:
I guess it's a matter of taste, but I would avoid importing generic names like `Collection` and `Item` in the global namespace like this.
I would just use `pystac.Collection()` and the like in the code, which is a bit more typing work, but a lot clearer reading-wise

(same with `Client` from `pystac_client`)

Reply:
Extra advantage of `import foo` over `from foo import bar`: it reduces risk on circular import issues